File: app/src/main/java/fr/nihilus/xenobladechronicles/MainActivity.java
package fr.nihilus.xenobladechronicles;

import android.arch.lifecycle.LifecycleRegistry;
import android.arch.lifecycle.LifecycleRegistryOwner;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;
import fr.nihilus.xenobladechronicles.monsters.list.MonsterListFragment;

public class MainActivity extends AppCompatActivity
        implements LifecycleRegistryOwner, HasSupportFragmentInjector {

    private final LifecycleRegistry mRegistry = new LifecycleRegistry(this);
    @Inject DispatchingAndroidInjector<Fragment> dispatchingFragmentInjector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // Injecte les dépendances dans les champs marqués par @Inject
        // Ceci doit être fait au plus tôt, avant que l'Activity soit complètement initialisée
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.container, new MonsterListFragment())
                    .commit();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        return true;
    }

    @Override
    public LifecycleRegistry getLifecycle() {
        return mRegistry;
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return dispatchingFragmentInjector;
    }
}

File: app/src/main/java/fr/nihilus/xenobladechronicles/MainActivity.kt
package fr.nihilus.xenobladechronicles

import android.os.Bundle
import android.view.Menu
import dagger.android.support.DaggerAppCompatActivity
import fr.nihilus.xenobladechronicles.monsters.list.MonsterListFragment

class MainActivity : DaggerAppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .add(R.id.container, MonsterListFragment())
                .commit()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean = true
}
